packages/editor-ui/src/views/NodeView.v2.vue

```diff
@@ -0,0 +1,985 @@
+const workflow = computed(() => workflowsStore.workflowsById[workflowId.value]);
+
+const editableWorkflow = computed(() => workflowsStore.workflow);
+const editableWorkflowObject = computed(() => workflowsStore.getCurrentWorkflow());
+
+const isDemoRoute = computed(() => route.name === VIEWS.DEMO);
+const isReadOnlyRoute = computed(() => route?.meta?.readOnlyCanvas === true);
+const isReadOnlyEnvironment = computed(() => {
+	return sourceControlStore.preferences.branchReadOnly;
+});
+
+const triggerNodes = computed<INodeUi[]>(() => {
+	return workflowsStore.workflowTriggerNodes;
+});
+
+const isCanvasAddButtonVisible = computed(() => {
+	return (
+		triggerNodes.value.length > 0 &&
+		!isLoading.value &&
+		!isDemoRoute.value &&
+		!isReadOnlyEnvironment.value
+	);
+});
+
+onMounted(() => {
+	void initialize();
+});
+
+async function initialize() {
+	isLoading.value = true;
+
+	const loadPromises: Array<Promise<unknown>> = [
+		nodeTypesStore.getNodeTypes(),
+		workflowsStore.fetchWorkflow(workflowId.value),
+	];
+
+	if (!settingsStore.isPreviewMode && !isDemoRoute.value) {
+		loadPromises.push(
+			workflowsStore.fetchActiveWorkflows(),
+			credentialsStore.fetchAllCredentials(),
+			credentialsStore.fetchCredentialTypes(true),
+		);
+
+		if (settingsStore.isEnterpriseFeatureEnabled(EnterpriseEditionFeature.Variables)) {
+			loadPromises.push(environmentsStore.fetchAllVariables());
+		}
+
+		if (settingsStore.isEnterpriseFeatureEnabled(EnterpriseEditionFeature.ExternalSecrets)) {
+			loadPromises.push(externalSecretsStore.fetchAllSecrets());
+		}
+	}
+
+	try {
+		await Promise.all(loadPromises);
+	} catch (error) {
+		return toast.showError(
+			error,
+			i18n.baseText('nodeView.showError.mounted1.title'),
+			i18n.baseText('nodeView.showError.mounted1.message') + ':',
+		);
+	}
+
+	initializeEditableWorkflow(workflowId.value);
+
+	if (window.parent) {
+		window.parent.postMessage(
+			JSON.stringify({ command: 'n8nReady', version: rootStore.versionCli }),
+			'*',
+		);
+	}
+
+	isLoading.value = false;
+}
+
+// @TODO Maybe move this to the store
+function initializeEditableWorkflow(id: string) {
+	const targetWorkflow = workflowsStore.workflowsById[id];
+
+	workflowsStore.addWorkflow(targetWorkflow);
+	workflowsStore.setWorkflow(targetWorkflow);
+	workflowsStore.setActive(targetWorkflow.active || false);
+	workflowsStore.setWorkflowId(targetWorkflow.id);
+	workflowsStore.setWorkflowName({ newName: targetWorkflow.name, setStateDirty: false });
+	workflowsStore.setWorkflowSettings(targetWorkflow.settings ?? {});
+	workflowsStore.setWorkflowPinData(targetWorkflow.pinData ?? {});
+	workflowsStore.setWorkflowVersionId(targetWorkflow.versionId);
+	workflowsStore.setWorkflowMetadata(targetWorkflow.meta);
+	if (targetWorkflow.usedCredentials) {
+		workflowsStore.setUsedCredentials(targetWorkflow.usedCredentials);
+	}
+
+	if (targetWorkflow.ownedBy) {
+		workflowsEEStore.setWorkflowOwnedBy({
```

Comment:
`setWorkflowOwnedBy` doesn't exist 🤔  Also, there's a lot of TS errors in this function due to invalid properties(`ownedBy`, `sharedBy`)

Reply:
Removed. Thank you!